pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestVisitor.java

```diff
@@ -160,10 +160,10 @@ public Void visitSort(SortNode node, ServerPlanRequestContext context) {
     if (node.getCollationKeys().size() > 0) {
       pinotQuery.setOrderByList(CalciteRexExpressionParser.convertOrderByList(node, pinotQuery));
     }
-    if (node.getFetch() > 0) {
+    if (node.getFetch() >= 0) {
```

Comment:
Probably can be simplified to:
```
pinotQuery.setLimit(node.getFetch());
pinotQuery.setOffset(node.getOffset());
```

Reply:
done.

Reply:
we actually need to keep this check, default value is -1 if null. Any number >=0 is explicitly passed from external.